### ISSUE TITLE
Review-Bot is failing. Can not obtain action secrets

### ISSUE THREAD

**Issue by Bullrich** [paritytech/polkadot-sdk#1754]
[Review bot](https://github.com/paritytech/polkadot-sdk/blob/master/.github/workflows/review-bot.yml) seems to be having an issue to obtain the repository secrets.

It has an environment set up for the secrets.

Related documentation: https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment

**Comment by Bullrich** [comment]
@altaua I created this ticket so we can centralize our progress.

I'll be experimenting on a staging repository to find the correct way to access variables.

**Comment by Bullrich** [comment]
I believe I found the issue.

It seems that `pull_request_target` counts as running an action outside of the branch, hence it doesn't allow it to access the environment secret.

I did a simple test and found the same result:
![image](https://github.com/paritytech/polkadot-sdk/assets/8524599/041aed48-901d-48d1-b500-10db2d9a8927)

This is why in #1748 the `pull_request_review` event is working but not the `pull_request_target`.

@pavelsupr I'm afraid we will have to remove the secret from the environment and put it as a repository secret for this to work.
